rust/src/table_state_arrow.rs

```diff
@@ -304,56 +304,81 @@ impl DeltaTableState {
             max_values: Option<ArrayRef>,
         }
 
+        let filter_out_empty_stats = |stats: &Result<ColStats, DeltaTableError>| -> bool {
+            let is_field_empty = |arr: &Option<ArrayRef>| -> bool {
+                match arr {
+                    Some(arr) => arr.len() == arr.null_count(),
+                    None => true,
+                }
+            };
+
+            let is_stats_empty = |stats: &ColStats| -> bool {
+                is_field_empty(&stats.null_count)
+                    && is_field_empty(&stats.min_values)
+                    && is_field_empty(&stats.max_values)
+            };
+
+            if let Ok(stats) = stats {
+                !is_stats_empty(stats)
+            } else {
+                true // keep errs
+            }
+        };
+
         let mut columnar_stats: Vec<ColStats> = SchemaLeafIterator::new(schema)
             .filter(|(_path, datatype)| !matches!(datatype, SchemaDataType::r#struct(_)))
             .map(|(path, datatype)| -> Result<ColStats, DeltaTableError> {
-                let null_count: Option<ArrayRef> = stats
+                let null_count = stats
                     .iter()
                     .flat_map(|maybe_stat| {
                         maybe_stat
                             .as_ref()
                             .map(|stat| resolve_column_count_stat(&stat.null_count, &path))
                     })
-                    .collect::<Option<Vec<DeltaDataTypeLong>>>()
-                    .map(arrow::array::Int64Array::from)
-                    .map(|arr| -> ArrayRef { Arc::new(arr) });
+                    .collect::<Vec<Option<DeltaDataTypeLong>>>();
+                let null_count: Option<ArrayRef> = empty_stats_into_null_array(Some(Arc::new(
+                    arrow::array::Int64Array::from(null_count),
+                )));
```

Comment:
We can save allocations if we detect the all-null case directly from `null_count`, rather than after we create the `Int64Array`. Same goes for the min and max below.

Reply:
Updated the PR